docs/features/logforwarding/parse-json-structured-logs.adoc

```diff
@@ -0,0 +1,152 @@
+*  `structuredTypeName`: (string, optional) Name to use for the structured type, unless _"structuredTypeKey"_ is set, and the value is present.
+
+==== Notes:
+* *At least one* of the structuredType options must be set when forwarding to an Elasticsearch type output
+* The Elasticsearch _index_ for structured records is formed by prepending "app-" to the structured type and appending "-write".
+* Unstructured records are not sent to the structured index, they are indexed as usual in application, infrastructure or audit indices.
+* If `structureTypeKey` is not found in the log entry, and no `structuredTypeName` is specified as a fallback, forward an _unstructured_ record without a `structured` field.
+
+
+==== Example
+
+For elasticsearch outputs, we must separate logs with different formats into different indices.
+Lets assume we have:
+
+* Applications logging in two structured JSON formats called "apache" and "google".
+* Each of our application pods have been labelled with the appropriate `logFormat=apache` or `logFormat=google`
+
+With the following forwarder configuration:
+
+.cluster-log-forwarder.yaml
+[source,yaml]
+----
+kind: ClusterLogForwarder
+apiVersion: logging.openshift.io/v1
+metadata:
+  name: instance
+  namespace: openshift-logging
+spec:
+  outputs:
+    - name: default
+      type: elasticsearch
+      elasticsearch:
+        structuredTypeKey: kubernetes.labels.logFormat
+  pipelines:
+    - name: my-formatted-logs
+      outputRefs:
+        - default
+      inputRefs:
+        - application
+      parse: json
+----
+. After parsing into _structured_, these log record will go to the index `app-apache-write`:
++
+[source,json]
+----
+{
+  "structured":{"name":"fred","home":"bedrock"},
+  "kubernetes":{"labels":{"logFormat": "apache", ...}}
+}
+----
+
+. This _structured_ log record will go to the index `app-google-write`:
++
+[source,json]
+----
+{
+  "structured":{"name":"wilma","home":"bedrock"},
+  "kubernetes":{"labels":{"logFormat": "google", ...}}
+}
+----
+
+*Note*: In the example above, only _structured_ logs that contain a `logFormat` label go to the `google` or `apache` index.
+All others go to the default application index as _unstructured_ records, including:
+
+* Records with missing or empty `logFormat` label.
+* Records that could not be parsed as JSON,  _even if_ they have a `logFormat` label.
+
+==== Elasticsearch Implementation Details
+It is *important* not to overload elasticsearch with too many indices.
+
+Only use distinct structured types for distinct log _formats_, *not* for each application or namespace.
+For example, most Apache applications use the same JSON log format and should use the same structured type, for example "LogApache".
+
+It is up to the user to ensure that logs sent to an index are consistent, otherwise index explosion and type confusion are still possible.
```

Comment:
what do we mean by "type confusion"

Reply:
changed to:

> It is up to the user to ensure that logs sent to an index are consistent.  Otherwise, it may result in index explosion and data inconsistency.